checkpoint/latest_version.go

```diff
@@ -123,7 +123,10 @@ func (lv *LatestVersion) Install(ctx context.Context) (string, error) {
 	if lv.ArmoredPublicKey != "" {
 		d.ArmoredPublicKey = lv.ArmoredPublicKey
 	}
-	err = d.DownloadAndUnpack(ctx, pv, dstDir)
+	zipFile, err := d.DownloadAndUnpack(ctx, pv, dstDir)
+	if zipFile != "" {
+		lv.pathsToRemove = append(lv.pathsToRemove, zipFile)
+	}
```

Comment:
```suggestion
	zipFilePath, err := d.DownloadAndUnpack(ctx, pv, dstDir)
	if zipFilePath != "" {
		lv.pathsToRemove = append(lv.pathsToRemove, zipFilePath)
	}
```
A minor nit-pick about naming - to make it clearer that this is a path.

It would be nice to avoid treating empty string as a magic value but I'm guessing most people would expect/want to do the cleanup even in the error case, e.g. [when the file size doesn't match or `io.Copy` fails](https://github.com/hashicorp/hc-install/blob/2f0f50e5b6aa573569b825a873b5dcadefb67b03/internal/releasesjson/downloader.go#L129-L138) for some reason, so I'm fine with this 👍🏻

Reply:
yeah that was exactly my thinking that we want to clean up even in the unhappy cases.